src/us/kbase/auth2/lib/identity/IdentityProviderConfig.java

```diff
@@ -125,10 +167,12 @@ public int hashCode() {
 		result = prime * result + ((clientID == null) ? 0 : clientID.hashCode());
 		result = prime * result + ((clientSecret == null) ? 0 : clientSecret.hashCode());
 		result = prime * result + ((customConfig == null) ? 0 : customConfig.hashCode());
+		result = prime * result + ((defaultLinkRedirectURL == null) ? 0 : defaultLinkRedirectURL.hashCode());
```

Comment:
Is there a reason that overriding the default hashCode function is needed for this class? Might be helpful to explain why in a comment

Reply:
Looks like you can't hash an object with nulls. Thats...unfortunate.

Reply:
> Is there a reason that overriding the default hashCode function is needed for this class? 

Because in Java, the default hash code is just the object address, and so is useless if you want equal object to hash to the same place.

> Might be helpful to explain why in a comment

This is totally standard java practice; a comment would be redundant. Also, this is auto generated code.

> Looks like you can't hash an object with nulls. Thats...unfortunate.

You can. Not sure why you think this?

Reply:
Thx. My Java knowledge is rudimentary, hence the stupid questions. I blame Steve.

Reply:
Note that `__hash__` should also be overridden in value classes in Python for the same reason, but people rarely do so. I mostly do it to make test code cleaner.